### PR TITLE
Delete hawkular metrics support

### PR DESCRIPTION
**Pull request by agrare**
Openshift 3.11 is the last version to support hawkular metrics, also supports prometheus metrics, and is going EOL in June 2022

### CODE REVIEW

File: manageiq-providers-kubernetes.gemspec [hunk 1]
```diff
@@ -18,7 +18,6 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "hawkular-client",                 "~> 5.0"
```
Comment (Fryguy) [inline manageiq-providers-kubernetes.gemspec:21]: 🎉

Comment (agrare) [reply manageiq-providers-kubernetes.gemspec:21]: Yeah, very happy about this particular line being deleted :laughing:

Comment (Fryguy) [reply manageiq-providers-kubernetes.gemspec:21]: For some reason dropping a dependency is my favorite part of any deletion; if the PR is designed to specifically delete a dependency, I'm like WAY happier about it.

cc @jrafanie 😉

Comment (agrare) [reply manageiq-providers-kubernetes.gemspec:21]: Yup hawkular-client was preventing us from updating something a while ago (rest-client I think?) and took forever to get the maintainers to cut a release for us.

Comment (jrafanie) [reply manageiq-providers-kubernetes.gemspec:21]: 🎉 One less dependency!!!  🥳

File: spec/models/manageiq/providers/kubernetes/container_manager/metrics_capture_spec.rb [hunk 2]
```diff
@@ -155,134 +74,5 @@
       @node.hardware.memory_mb = nil
       expect { @node.perf_collect_metrics('interval_name') }.to raise_error(described_class::TargetValidationError)
     end
-
-    # TODO: include also sort_and_normalize in the tests
-    METRICS_EXERCISES = [
```
Comment (agrare) [inline spec/models/manageiq/providers/kubernetes/container_manager/metrics_capture_spec.rb:160]: NOTE we already have VCR specs for prometheus that cover these methods

**Comment by Fryguy** [comment]
Do we need a data migration to drop any authentication records?

**Comment by agrare** [comment]
@Fryguy yes I think we do

**Pull request merged**